controller_manager:
  ros__parameters:
    update_rate: 30
    use_sim_time: true

    diffdrive_controller:
      type: diff_drive_controller/DiffDriveController

    joint_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

diffdrive_controller:
  ros__parameters:
    publish_rate: 50.0

    base_frame_id: base_link

    left_wheel_names: ['base_Lwheel_joint']
    right_wheel_names: ['base_rwheel_joint']
    wheel_separation: 0.228
    wheel_radius: 0.028
    
    use_stamped_vel: false
    
    # wheel_separation_multiplier: 1.0
    # left_wheel_radius_multiplier: 1.0
    # right_wheel_radius_multiplier: 1.0

    # odom_frame_id: odom
    # base_frame_id: base_link
    # pose_covariance_diagonal: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    # twist_covariance_diagonal: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]

    # position_feedback: false
    # open_loop: true
    # enable_odom_tf: true

    # cmd_vel_timeout: 0.5 # seconds
    # publish_limited_velocity: true
    # velocity_rolling_window_size: 10

    # linear.x.has_velocity_limits: false
    # linear.x.has_acceleration_limits: false
    # linear.x.has_jerk_limits: false
    # linear.x.max_velocity: 0.0
    # linear.x.min_velocity: 0.0
    # linear.x.max_acceleration: 0.0
    # linear.x.max_jerk: 0.0
    # linear.x.min_jerk: 0.0

    # angular.z.has_velocity_limits: false
    # angular.z.has_acceleration_limits: false
    # angular.z.has_jerk_limits: false
    # angular.z.max_velocity: 0.0
    # angular.z.min_velocity: 0.0
    # angular.z.max_acceleration: 0.0
    # angular.z.min_acceleration: 0.0
    # angular.z.max_jerk: 0.0
    # angular.z.min_jerk: 0.0